src/js/controllers/purchase.component.js

```diff
@@ -0,0 +1,177 @@
+            this.#renderToDeleteInput();
+        }
+    }
+
+    _submitByEnterKey(e) {
+        if (e.key !== 'Enter') return;
+        e.preventDefault();
+        this.renderManuelLotto();
+    }
+
+    renderManuelLotto() {
+        if (!this.#isInputChange) return;
+        this._stateModel.setState('reset', true);
+
+        const params = {
+            sectionType: SECTIONTYPE.PURCHASE,
+            value: $purchased.amount.value
+        }
+        if (!this._validator.validate(params)) return;
+
+        this._stateModel.setState('price', $purchased.amount.value);
+        this._stateModel.setState('purchasedUnit', $purchased.amount.value / PRICE_PER_UNIT);
+        this._view.displayBlock([$purchasedManuel.lotto]);
+        this.#isInputChange = false;
+        this.#setDisabledDeleteButton();
+    }
+
+    purchase() {
+        if (this._stateModel.getState('purchased') || !this.#isValidated()) return;
+
+        this._stateModel.setState('purchased', true);
+        if (this.#hasNumberSetManuel()) {
+            this._stateModel.setState('numberSetManuel', this.numberSetManuel);
+        }
+
+        this._view.renderDisabledButton([
+            $purchasedManuel.button,
+            $purchasedManuel.addButton,
+            $purchasedManuel.deleteButton
+        ], true);
+
+        new IssueComponent({
+            view: this._view,
+            state: this._stateModel,
+            validator: this._validator
+        });
+    }
+
+    #renderToAddInput() {
+        const params = {
+            sectionType: SECTIONTYPE.MANUEL_INPUT,
+            length: $purchasedManuel.set.children.length,
+            unit: $purchased.amount.value / PRICE_PER_UNIT
+        }
+        if (!this._validator.validate(params)) return this.#renderToDeleteInput();
+        this._view.renderToAppend($purchasedManuel.set, stringParserToHTML($lottoManuel));
+        this.#setDisabledDeleteButton();
+        $purchasedManuel.inputs = this._view.renderToUpdateSelector({
+            selector: $purchasedManuel.inputs,
+            className: '.purchased-lotto-manuel-inputs',
+            isAll: true
+        });
+    }
+
+    #renderToDeleteInput() {
+        const $lastLottoElement = $purchasedManuel.set.childNodes[$purchasedManuel.set.children.length - 1];
+        this._view.remove($lastLottoElement);
+        this.#setDisabledDeleteButton();
+    }
+
+    #setDisabledDeleteButton() {
+        if ($purchasedManuel.set.children.length < 2) {
+            this._view.renderDisabledButton([$purchasedManuel.deleteButton], true);
+            return;
+        }
+
+        return this._view.renderDisabledButton([$purchasedManuel.deleteButton], false);
+    }
+
+    #setAutoFocus(e) {
+        const $InputNextSibling = e.target.nextElementSibling;
+        const isValueMaxLength = e.target.maxLength === e.target.value.length;
+        if (!!$InputNextSibling && isValueMaxLength) {
+            this._view.renderToSetFocus($InputNextSibling);
+        }
```

Comment:
(마이너) 명시적으로 조건을 표현하는게 더 이해하기 편할거같은데 어떻게 생각하실까요? 
```suggestion
        if ($InputNextSibling != null && isValueMaxLength) {
            this._view.renderToSetFocus($InputNextSibling);
        }
```

Reply:
이제껏 저의 개발 습관으로 !!으로 대체될 수있는 조건들은 간결하게 !!연산자로 표현해왔는데, 미션을 진행하면서 리뷰어님들의 조언을 들으면서 조건식을 사용할 때 명확한 조건을 보여주어야 한다는게 코드를 읽을 때 이해하기 더 쉽겠구나 라는 걸 깨닫게 되네요! 요부분 수정하도록 하겠습니다!